code/model/modelread.cpp

```diff
@@ -3322,18 +3373,16 @@ void model_find_obj_dir(vec3d *w_vec, vec3d *m_vec, object *pship_obj, int sub_m
 	vm_vec_unrotate(w_vec, &vec, &pship_obj->orient);
 }
 
-void model_instance_find_obj_dir(vec3d *w_vec, vec3d *m_vec, object *pship_obj, int sub_model_num)
+void model_instance_find_obj_dir(vec3d *w_vec, vec3d *m_vec, matrix *orient, int model_num, int model_instance_num, int submodel_num)
 {
 	vec3d tvec, vec;
 	matrix m;
 	int mn;
 
-	Assert(pship_obj->type == OBJ_SHIP);
-
-	polymodel_instance *pmi = model_get_instance(Ships[pship_obj->instance].model_instance_num);
-	polymodel *pm = model_get(Ship_info[Ships[pship_obj->instance].ship_info_index].model_num);
+	polymodel_instance *pmi = model_get_instance(model_instance_num);
+	polymodel *pm = model_get(model_num);
```

Comment:
An Assertion that checks if the model instance actually belongs to the model number could catch some programming errors here.

On seconds thought, `polymodel_instance` contains the model number of the referenced `polymodel` so the `model_num` parameter isn't needed at all.